File: ml-service/app/schemas/attribution.py
from pydantic import BaseModel


class FeatureImpact(BaseModel):
    feature: str
    impact: float


class PredictionExplanation(BaseModel):
    station: str
    timestamp: str

    actual_aqi: float
    predicted_aqi: float
    prediction_error: float

    prediction_correct: bool

    top_features: list[FeatureImpact]


class FeatureImportance(BaseModel):
    feature: str
    importance: float

File: frontend/src/types/attribution.ts
export interface FeatureImpact {
  feature: string;
  impact: number;
}

export interface AttributionResponse {
  station: string;
  timestamp: string;

  actual_aqi: number;
  predicted_aqi: number;

  prediction_error: number;
  prediction_correct: boolean;

  top_features: FeatureImpact[];
}

export interface GlobalFeatureImportance {
  feature: string;
  importance: number;
}

export interface FeatureImpact {
  feature: string;
  impact: number;
}

export interface PredictionExplanation {
  station: string;
  timestamp: string;

  actual_aqi: number;

  predicted_aqi: number;

  prediction_error: number;

  prediction_correct: boolean;

  top_features: FeatureImpact[];
}